### ISSUE TITLE
allow callback for sqlite file config 

### ISSUE THREAD

**Issue by bnomei** [distantnative/search-for-kirby#20]
```
'sqlite' => [
        'file'  => ..., // absolute path, default points to `media/search.sqlite`
    ]
```

the default is actually not `media/search.sqlite` but the absolute path `DISK/SOME/FOLDERS/media/search.sqlite`.

it is possible to set the path using `dirname`
```php
'file'  => dirname(__DIR__, 1) . '/search/search.sqlite', // /site/config/config.php => /site/search/search.sqlite
```

but using an optional callback might be a bit more readable
```php
'file'  => function() {
   return kirby()->roots()->index() . '/site/search/search.sqlite',
}
```

**Comment by distantnative** [comment]
> the default is actually not media/search.sqlite but the absolute path DISK/SOME/FOLDERS/media/search.sqlite.

That's true but I don't really think there is any absolute example path that wouldn't confuse other users as well - since absolute paths are very dependent on your own setup. But that's why it says "absolute" path right at the beginning ;)

Regarding the callback idea, yes that sounds like this could work.

**Comment by distantnative** [comment]
✅ Will be included in 1.0.0 release

**Comment by bnomei** [comment]
about the configurable absolute paths... in my plugins i try to avoid them and use callbacks and kirbys roots as much as possible since i think it makes them more readable and reliable. just my personal taste.

thanks for adding the support for callback. 👏
